docs/conceptual/node/tx-pool.md

```diff
@@ -2,12 +2,67 @@
+
+`Priority` helps define the linear ordering of a graph in the case of one transaction unlocking multiple dependent transactions. For example, if we have two (or more) transactions that have their dependencies satisfied, then we use `priority` to choose the order for them.
+
+
+For `Frame` based runtimes, `priority` is defined as the `fee` that the transaction is going to pay. For example, if we receive two transactions from the same sender with an identical `nonce`, only one transaction can be included on-chain. We use `priority` to choose the transaction with a higher `fee` to store in the transaction pool.
+
+Note that the pool does not know about fees, accounts or signatures - it only deals with the validity of the transaction and the abstract notion of `priority`, `requires` and `provides` parameters.
+
+## Transaction Lifecycle
+
+There are two paths an extrinsic can follow:
+
+1. **A block is received from the network**
+
+  The block is run through `execute_block`([code](https://github.com/paritytech/substrate/blob/20a9b15cdbed4bf962a4447e8bfb812f766f2fbc/frame/executive/src/lib.rs#L196)), and the entire block either succeeds or fails.
+
+2. **A block is produced by our own node**
+
+  a) Our node listens for transactions on the network.
+
+  b) Each transaction is verified through the `validate_transaction` ([code](https://github.com/paritytech/substrate/blob/20a9b15cdbed4bf962a4447e8bfb812f766f2fbc/frame/executive/src/lib.rs#L304)) function and places valid transactions in the transaction pool.
+
+  c) The pool is responsible for ordering the transactions and returning ones that are ready to be included in the block. Transactions in the ready queue are used to construct a block with `BlockBuilder`.
+
+  d) `BlockBuilder`([code](https://github.com/paritytech/substrate/blob/master/client/block-builder/src/lib.rs)) uses the `apply_extrinsic`([code](https://github.com/paritytech/substrate/blob/20a9b15cdbed4bf962a4447e8bfb812f766f2fbc/frame/executive/src/lib.rs#L235)) function to execute the transaction and apply the state changes in local memory.
+
+  e) The constructed block is published to the network (where all other nodes run it through `execute_block`).
+
+## Transaction Validity
+
+`validate_transaction` is called from the runtime, checks for a valid signature and nonce (or output for a UTXO chain) and returns a [`Result`](https://github.com/paritytech/substrate/blob/20a9b15cdbed4bf962a4447e8bfb812f766f2fbc/primitives/runtime/src/transaction_validity.rs#L150). `validate_transaction` checks transactions in isolation, so it will not catch errors like the same output being spent twice.
+
+Although it is possible, `validate_transaction` does not check if calls to modules will succeed. It is a potential DoS vector since all transactions in the network will be passed into `validate_transaction`.
```

Comment:
```suggestion
Although it is possible, `validate_transaction` does not check whether calls to pallets will succeed. It is a potential DoS vector since all transactions in the network will be passed into `validate_transaction`.
```